skyportal/models/photometry.py

```diff
@@ -248,3 +263,720 @@ def snr(self):
+        self.check_path_string(self.series_obj_id)
+
+        # we can let series_identifier have slashes, which makes subdirs
+        self.check_path_string(self.series_identifier, allow_slashes=True)
+
+        # make sure to replace windows style slashes
+        subfolder = self.series_identifier.replace("\\", "/")
+
+        filename = f'photo_series_{self.series_obj_id}.nc'
+
+        path = os.path.join(root_folder, subfolder)
+        if not os.path.exists(path):
+            os.makedirs(path)
+
+        full_name = os.path.join(path, filename)
+
+        if len(full_name) > MAX_FILEPATH_LENGTH:
+            raise ValueError(
+                f'Full path to file {full_name} is longer than {MAX_FILEPATH_LENGTH} characters.'
+            )
+
+        self._data.to_netcdf(full_name)
+
+        self.filename = full_name
+
+    def delete_data(self):
+        """
+        Delete the underlying photometric data from disk
+        """
+
+        if os.path.exists(self.filename):
+            os.remove(self.filename)
+
+    read = (
+        accessible_by_groups_members
+        | accessible_by_streams_members
+        | accessible_by_owner
+    )
+    update = delete = accessible_by_owner
+
+    filename = sa.Column(
+        sa.String,
+        nullable=False,
+        index=True,
+        doc="Full path and filename, or URI to the netCDF file storing photometric data.",
+    )
+
+    mjd_first = sa.Column(
+        sa.Float,
+        nullable=False,
+        doc='MJD of the first exposure of the series.',
+        index=True,
+    )
+
+    mjd_mid = sa.Column(
+        sa.Float,
+        nullable=False,
+        doc='MJD of the middle of the observation series.',
+        index=True,
+    )
+
+    mjd_last = sa.Column(
+        sa.Float,
+        nullable=False,
+        doc='MJD of the last exposure of the series.',
+        index=True,
+    )
+
+    mjd_last_detected = sa.Column(
+        sa.Float,
+        nullable=False,
+        doc='MJD of the last exposure that was above threshold.',
+        index=True,
+    )
+
+    detected = sa.Column(
+        sa.Boolean,
+        nullable=False,
+        doc='True if any of the data points are above threshold.',
+        index=True,
+    )
+
+    series_identifier = sa.Column(
+        sa.String,
+        nullable=False,
+        doc='Unique identifier of the series of images out of which the photometry is generated.',
+        index=True,
+    )
+
+    series_obj_id = sa.Column(
+        sa.String,
+        nullable=False,
+        doc='Unique identifier of an object inside the series of images out of which the photometry is generated.',
+    )
+
+    channel_id = sa.Column(
+        sa.String, nullable=False, default='0', doc='Channel of the photometric series.'
+    )
+
+    filter = sa.Column(
+        allowed_bandpasses,
+        nullable=False,
+        doc='Filter with which the observation was taken.',
+    )
+
+    exp_time = sa.Column(
```

Comment:
if exp_time is missing in the data then this will be None -- it should be nullable